File: user interface/src/src/store/payloads/add-prescription-payload.ts
import TimeOfDay from "../domain/time-of-day";
import moment from 'moment';

class AddPrescriptionPayload {
  constructor(){
    this.name = '';
    this.unitDose = '';
    this.startDate = moment().toISOString()
    this.timesOfDay = [
      new TimeOfDay(0, 0)
    ]
  }

  name: string;
  unitDose: string;
  startDate: string;
  completeDate?: string;
  totalQuantity?: number;
  timesOfDay: TimeOfDay[]
}

export default AddPrescriptionPayload;

File: services/src/services.UserPrescriptions/Domain/AddPrescriptionPayload.cs
using System;
using System.Collections.Generic;

namespace services.UserPrescriptions.Domain
{
    public class AddPrescriptionPayload
    {
        public AddPrescriptionPayload()
        {
            TimesOfDay = new List<TimeOfDay>();
        }
        public string Name { get; set; }
        public string UnitDose { get; set; }
        public int TotalQuantity { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? CompleteDate { get; set; }
        public List<TimeOfDay> TimesOfDay { get; set; }
    }
}